test/cluster_test.go

```diff
@@ -463,12 +463,14 @@ func TestProperFalloutAfterMaxAttemptsWithAuthMismatch(t *testing.T) {
 
 	// Wait for ClosedCB
 	if e := WaitTime(cch, 5*time.Second); e != nil {
-		t.Fatalf("Did not receive a closed callback message, #reconnects: %v", nc.Reconnects)
+		reconnects := nc.Stats().Reconnects
+		t.Fatalf("Did not receive a closed callback message, #reconnects: %v", reconnects)
 	}
 
 	// Make sure we have not exceeded MaxReconnect
 	if nc.Reconnects != uint64(opts.MaxReconnect) {
```

Comment:
You could still have the race here then. Move the capture of `nc.Stats().Reconnects` in the if statement.